echidna-lb/src/config/mod.rs

```diff
@@ -0,0 +1,47 @@
+use clap::Parser;
+use serde::Deserialize;
+use serde_yaml::Error;
+use std::fs;
+use std::path::Path;
+
+#[derive(Parser)]
+#[clap(author, version, about, long_about = None)]
+pub struct Args {
+    /// Path to the configuration file
+    #[clap(short, long, default_value = "./config.yaml")]
+    pub config: String,
+}
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub port: Option<u16>,
+    pub debug: Option<bool>,
+    pub https_port: Option<u16>,
+    pub algorithm: String,
+    pub healthcheck: Option<HealthcheckConfig>,
+    pub backends: Vec<BackendConfig>,
+    pub ssl: Option<SslConfig>,
+}
+
+#[derive(Deserialize)]
+pub struct HealthcheckConfig {
+    pub interval_sec: u64,
+    pub route: String,
+}
+
+#[derive(Deserialize)]
+pub struct BackendConfig {
+    pub url: String,
+    pub weight: usize,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct SslConfig {
+    pub cert_path: String,
+    pub key_path: String,
+}
+
+pub fn load_config<P: AsRef<Path>>(path: P) -> Result<Config, Error> {
+    let config_str = fs::read_to_string(path).expect("Unable to read config file");
```

Comment:
This will panic, it's bad practice to panic in a function which returns an error. Should return an error IMO.

Reply:
Handled errors in all the places.